pkg/api/v1beta3/types.go

```diff
@@ -1641,3 +1641,34 @@ type SecretList struct {
 
 	Items []Secret `json:"items" description:"items is a list of secret objects"`
 }
+
+// Type and constants for component health validation.
+type ComponentConditionType string
+
+// These are the valid conditions for the component.
+const (
+	ComponentHealthy ComponentConditionType = "Healthy"
+)
+
+type ComponentCondition struct {
+	Type    ComponentConditionType `json:"type"`
+	Status  ConditionStatus        `json:"status"`
+	Message string                 `json:"message,omitempty" description:"health check message received from the component"`
+	Error   string                 `json:"error,omitempty" description:"error code from health check attempt (if any)"`
+}
+
+// ComponentStatus (and ComponentStatusList) holds the cluster validation info.
+type ComponentStatus struct {
```

Comment:
I think having a Component object eventually is fine. ComponentStatus is also very much intended to be a virtual resource -- it shouldn't be persisted anywhere. The trick is that who gets to be called a component and how it gets registered. Right now the list is a fixed list, with fixed end points it hits.